pyface/ui/qt4/tasks/advanced_editor_area_pane.py

```diff
@@ -61,29 +69,48 @@ def create(self, parent):
             prev_seq = "Ctrl+PgUp"
         shortcut = QtGui.QShortcut(QtGui.QKeySequence(next_seq), self.control)
         shortcut.activated.connect(self._next_tab)
+        self._connections_to_remove.append(
+            (shortcut.activated, self._next_tab)
+        )
         shortcut = QtGui.QShortcut(QtGui.QKeySequence(prev_seq), self.control)
         shortcut.activated.connect(self._previous_tab)
+        self._connections_to_remove.append(
+            (shortcut.activated, self._previous_tab)
+        )
 
         # Add shortcuts for switching to a specific tab.
         mod = "Ctrl+" if sys.platform == "darwin" else "Alt+"
         mapper = QtCore.QSignalMapper(self.control)
         mapper.mapped.connect(self._activate_tab)
+        self._connections_to_remove.append(
+            (mapper.mapped, self._activate_tab)
+        )
         for i in range(1, 10):
             sequence = QtGui.QKeySequence(mod + str(i))
             shortcut = QtGui.QShortcut(sequence, self.control)
             shortcut.activated.connect(mapper.map)
+            self._connections_to_remove.append(
+                (shortcut.activated, mapper.map)
+            )
             mapper.setMapping(shortcut, i - 1)
 
     def destroy(self):
         """ Destroy the toolkit-specific control that represents the pane.
         """
         self.control.removeEventFilter(self._filter)
+        self.control._remove_event_listeners()
         self._filter = None
 
         for editor in self.editors:
             editor_widget = editor.control.parent()
             self.control.destroy_editor_widget(editor_widget)
             editor.editor_area = None
+        self.active_editor = None
+
+        if self.control is not None:
```

Comment:
The first line in this function already accesses `control.*` so maybe this `if self.control is not None` is not needed? (See another comment too).